shopelectro/views/catalog.py

```diff
@@ -7,75 +7,93 @@
 from django.views.decorators.http import require_POST
 from django_user_agents.utils import get_user_agent
 
+# can't do `import pages` because of django error.
+# Traceback: https://gist.github.com/duker33/685e8a9f59fc5dbd243e297e77aaca42
+from pages import models as pages_models, views as pages_views
 from catalog import newcontext
 from catalog.views import catalog
 from images.models import Image
-from pages import views as pages_views
 from shopelectro import models, context as se_context
+from shopelectro import request_data
 from shopelectro.views.helpers import set_csrf_cookie
 
-PRODUCTS_ON_PAGE_PC = 48
-PRODUCTS_ON_PAGE_MOB = 12
 
+class CatalogContext(newcontext.Context):
```

Comment:
let's move this class to `shopelectro.context`

Reply:
Let's move `category`, `get_all_tags`, `filter_products` and `paginate_products` to `context` method as local variables.
They are zero-logic and almost declarative methods, which just make a noise and grow the size of `CatalogContext` in form of methods